Extensions/PathfindingBehavior/tests/commonpathfindingruntimebehavior.spec.js

```diff
@@ -0,0 +1,161 @@
+  allowDiagonals
+) => {
+  var runtimeGame = new gdjs.RuntimeGame({
+    variables: [],
+    properties: { windowWidth: 800, windowHeight: 600 },
+    resources: { resources: [] },
+  });
+  var runtimeScene = new gdjs.RuntimeScene(runtimeGame);
+  runtimeScene.loadFromScene({
+    layers: [{ name: '', visibility: true, effects: [] }],
+    variables: [],
+    behaviorsSharedData: [],
+    objects: [],
+    instances: [],
+  });
+  runtimeScene._timeManager.getElapsedTime = function () {
+    return (1 / 60) * 1000;
+  };
+
+  const pathFindingName = 'auto1';
+  var object = new gdjs.RuntimeObject(runtimeScene, {
+    name: 'player',
+    type: '',
+    behaviors: [
+      {
+        type: 'PathfindingBehavior::PathfindingBehavior',
+        name: pathFindingName,
+        allowDiagonals: allowDiagonals,
+        acceleration: 400,
+        maxSpeed: 200,
+        angularMaxSpeed: 180,
+        rotateObject: false,
+        angleOffset: 0,
+        cellWidth: 20,
+        cellHeight: 20,
+        extraBorder: 0,
+      },
+    ],
+  });
+  object.getWidth = function () {
+    return 90;
+  };
+  object.getHeight = function () {
+    return 90;
+  };
+  runtimeScene.addObject(object);
+
+  it('can find a path without any obstacle at all', function () {
+    object.setPosition(480, 300);
+    object.getBehavior(pathFindingName).moveTo(runtimeScene, 720, 300);
+    expect(object.getBehavior(pathFindingName).pathFound()).to.be(true);
+    expect(object.getBehavior(pathFindingName).getNodeCount()).to.be(13);
+  });
+
+  it('can find a path without any obstacle in the way', function () {
+    const obstacle = addObstacle(runtimeScene);
+    obstacle.setPosition(100, 100);
+    // so obstacles register
+    runtimeScene.renderAndStep(1000 / 60);
+
+    object.setPosition(480, 300);
+    object.getBehavior(pathFindingName).moveTo(runtimeScene, 720, 300);
+    expect(object.getBehavior(pathFindingName).pathFound()).to.be(true);
+    expect(object.getBehavior(pathFindingName).getNodeCount()).to.be(13);
+  });
+
+  const obstacle = addObstacle(runtimeScene);
+
+  it("mustn't find a path to the obstacle inside", function () {
+    obstacle.setPosition(600, 300);
+    // so obstacles register
+    runtimeScene.renderAndStep(1000 / 60);
+
+    object.setPosition(480, 300);
+    object.getBehavior(pathFindingName).moveTo(runtimeScene, 650, 350);
+    expect(object.getBehavior(pathFindingName).pathFound()).to.be(false);
+  });
+
+  it('can find a path with an obstacle in the way', function () {
+    obstacle.setPosition(600, 300);
+    // so obstacles register
+    runtimeScene.renderAndStep(1000 / 60);
+
+    object.setPosition(480, 300);
+    object.getBehavior(pathFindingName).moveTo(runtimeScene, 720, 300);
+    expect(object.getBehavior(pathFindingName).pathFound()).to.be(true);
+    expect(object.getBehavior(pathFindingName).getNodeCount()).to.be.above(13);
+  });
+
+  const obstacleTop = obstacle;
+  const obstacleBottom = addObstacle(runtimeScene);
```

Comment:
In general try to avoid declaring anything between tests. Tests are supposed to be independent, so the last one could be launched before the first one, vice versa, etc...

Prefer to set up the scene and obstacles for each test (you can make a helper function inside the function if needed, to avoid too much code duplication :)).

In other words, a test should be self contained - this will allow them to run in parallel and to avoid weird stuff like "my test is failing because the previous test failed, now I'm confused" ;)